runtime/src/main/java/org/corfudb/protocols/wireprotocol/NettyCorfuMessageEncoder.java

```diff
@@ -1,29 +1,66 @@
 package org.corfudb.protocols.wireprotocol;
 
 import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufOutputStream;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.MessageToByteEncoder;
-
+import java.io.IOException;
+import java.util.concurrent.atomic.LongAccumulator;
 import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.proto.service.CorfuMessage.RequestMsg;
+import org.corfudb.runtime.proto.service.CorfuMessage.ResponseMsg;
 
-import java.util.concurrent.atomic.LongAccumulator;
+import static org.corfudb.protocols.CorfuProtocolCommon.LEGACY_CORFU_MSG_MARK;
+import static org.corfudb.protocols.CorfuProtocolCommon.PROTO_CORFU_REQUEST_MSG_MARK;
+import static org.corfudb.protocols.CorfuProtocolCommon.PROTO_CORFU_RESPONSE_MSG_MARK;
 
 /**
  * Created by mwei on 10/1/15.
  */
 @Slf4j
-public class NettyCorfuMessageEncoder extends MessageToByteEncoder<CorfuMsg> {
+public class NettyCorfuMessageEncoder extends MessageToByteEncoder<Object> {
 
 
     final LongAccumulator maxValue = new LongAccumulator(Math::max, Long.MIN_VALUE);
 
     @Override
-    protected void encode(ChannelHandlerContext channelHandlerContext,
-                          CorfuMsg corfuMsg,
-                          ByteBuf byteBuf) throws Exception {
+    protected void encode(ChannelHandlerContext channelHandlerContext, Object object, ByteBuf byteBuf) {
```

Comment:
can we consider adding the missing java doc for this method?

Reply:
Done.